e2e/scripts/drivers.js

```diff
@@ -0,0 +1,63 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import {promisify} from 'util';
+
+import {DRIVERS_DIR} from './constants';
+
+export class Driver {
+  constructor({browser, url, driverFilename}) {
+    this.browser = browser;
+    this.url = url;
+    this._driverFilename = driverFilename;
+  }
+
+  get downloadFilename() {
+    const urlPathSegments = this.url.split('/');
+    return urlPathSegments[urlPathSegments.length - 1];
+  }
+
+  get driverFilename() {
+    return this._driverFilename || this.downloadFilename;
+  }
+
+  get pathTo() {
+    return path.join(DRIVERS_DIR, this.driverFilename);
+  }
+}
+
+export function getOsName() {
+  return {
+    Darwin: 'mac',
+    Linux: 'linux',
+    Windows_NT: 'win'
+  }[os.type()];
+}
+
+export function getOsArch() {
+  return os.arch().slice(1);
+}
+
+export function addDriversToPATH() {
+  process.env.PATH = `${process.env.PATH}:${DRIVERS_DIR}`;
+}
+
+export const DRIVERS = [
+  new Driver({
+    browser: 'chrome',
+    url: `https://chromedriver.storage.googleapis.com/2.33/chromedriver_${getOsName()}${getOsArch()}.zip`,
+    driverFilename: 'chromedriver'
+  })
+];
+
+export async function hasDrivers() {
+  try {
+    const realDriverFilenames = new Set(await promisify(fs.readdir)(DRIVERS_DIR));
+    return !DRIVERS.find((driver) => !realDriverFilenames.has(driver.driverFilename));
```

Comment:
I'm always on the lookout to good solutions to brain fry!

I believe the double negative here is necessary, because if you think about it, it's not a _real_ double negative:

The inner `!realDriverFilenames.has(driver.driverFilename)`: We iterate over the `DRIVERS` array, looking for one that is _not_ downloaded. If we got rid of this, we might not make it through the whole array, and may miss a missing driver.

The outer `!DRIVERS.find(...)`: if that inner find returns us a driver, then we need to download that webdriver; if it returns `undefined`, then we've got all our drivers. The truthiness of those values is the opposite of the question "do we have all of our drivers?"

Hope that helps!